pallets/ds-maps/src/lib.rs

```diff
@@ -0,0 +1,183 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+#[cfg(feature = "std")]
+use serde::{Deserialize, Serialize};
+
+use frame_support::{
+    codec::{Decode, Encode},
+    decl_error, decl_event, decl_module, decl_storage, dispatch, ensure,
+    weights::{Weight},
+    Parameter,
+};
+use frame_system::ensure_signed;
+use pallet_ds_accounts as accounts;
+use accounts::REGISTRAR_ROLE;
+
+mod default_weight;
+#[cfg(test)]
+mod mock;
+#[cfg(feature = "payment")]
+mod payment;
+#[cfg(test)]
+mod tests;
+
+//Not sure, are those derives required?
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Encode, Decode, Clone, PartialEq, Eq, Debug)]
+pub enum ZoneType {
+    /// Forbidden type zone
+    Red,
+    /// Available for safe flights
+    Green,
+    /// Owns zones
+    Parent,
+    // 
+}
+
+impl Default for ZoneType {
+    fn default() -> Self {
+        ZoneType::Green
+    }
+}
+
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Encode, Decode, Clone, Default, Debug, PartialEq, Eq)]
+pub struct BoxCoordinates<CoordinateSize> {
+    coordinates: [CoordinateSize; 6],
+}
+impl<CoordinateSize> BoxCoordinates<CoordinateSize>{
+    pub fn new(coordinates: [CoordinateSize; 6]) -> Self {
+        BoxCoordinates{coordinates}
+    }
+}
+
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Encode, Decode, Clone, Default)]
+pub struct Zone<CoordinateSize> {
+    pub coordinates: BoxCoordinates<CoordinateSize>,
+    pub zone_type: ZoneType,
+    pub zone_id: u32,
+}
+
+impl<CoordinateSize> Zone<CoordinateSize>
+{
+    pub fn zone_is(&self, zone: ZoneType) -> bool{
+        !(self.zone_type == zone)
+    }
+    pub fn new( zone_id: u32, 
+                zone_type: ZoneType, 
+                coordinates: BoxCoordinates<CoordinateSize> ) -> Self {
+            Zone {
+                coordinates,
+                zone_type,
+                zone_id,
+            }
+    }
+}
+
+
+/// Configure the pallet by specifying the parameters and types on which it depends.
+pub trait Trait: accounts::Trait {
+    /// Because this pallet emits events, it depends on the runtime's definition of an event.
+    type Event: From<Event<Self>> + Into<<Self as frame_system::Trait>::Event>;
+    // Describe pallet constants.
+    // Lean more https://substrate.dev/docs/en/knowledgebase/runtime/metadata
+    type WeightInfo: WeightInfo;
+    // new types, fixes required
+    type Coordinates: Parameter;
+    ///guess use u32 for representing global coords, u16 for local
+    type CoordinateSize: Default + Parameter;
```

Comment:
What's the difference? Why do we need 2 types for coordinates?

Reply:
Changed naming, now understandable.